sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1118,6 +1126,100 @@ ProgramManager::KernelArgMask ProgramManager::getEliminatedKernelArgMask(
   return {};
 }
 
+static bundle_state getBinImageState(RTDeviceBinaryImage *BinImage) {
+  auto IsAOTBinary = [](const char *Format) {
+    if ((strcmp(Format, __SYCL_PI_DEVICE_BINARY_TARGET_SPIRV64_X86_64) == 0) ||
+        (strcmp(Format, __SYCL_PI_DEVICE_BINARY_TARGET_SPIRV64_GEN) == 0) ||
+        (strcmp(Format, __SYCL_PI_DEVICE_BINARY_TARGET_SPIRV64_FPGA) == 0))
+      return true;
+    return false;
+  };
+
+  // There are only two initial states so far - SPIRV which needs to be compiled
+  // and linked and fully compiled(AOTed) binary
+
+  const bool IsAOT = IsAOTBinary(BinImage->getRawData().DeviceTargetSpec);
+
+  return IsAOT ? sycl::bundle_state::executable : sycl::bundle_state::input;
+}
+
+static bool compatibleWithDevice(RTDeviceBinaryImage *BinImage,
+                                const device &Dev) {
+  const std::shared_ptr<detail::device_impl> &DeviceImpl =
+      detail::getSyclObjImpl(Dev);
+  auto &Plugin = DeviceImpl->getPlugin();
+
+  const RT::PiDevice &PIDeviceHandle = DeviceImpl->getHandleRef();
+
+  // Call piextDeviceSelectBinary with only one image to check if an image is
+  // compatible with implementation. The function returns invalid index if no
+  // device images are compatible.
+  pi_uint32 SuitableImageID = 42;
```

Comment:
Do you mind if I initialize it?

Reply:
Actually, my suggestion goes against C++ core guidelines, so disregard this.

Reply:
Can something like `std::numeric_limits<pi_uint32>::max()` be used instead of `42`? Otherwise reader may suspect there is some hidden meaning behind `42`.